app/resources/api/v2/custom_metadatum_collection_resource.rb

```diff
@@ -5,31 +5,41 @@ module V2
     # Provides a JSON API representation of custom_metadatum_collection
     # See: http://jsonapi-resources.com/ for JSONAPI::Resource documentation
     class CustomMetadatumCollectionResource < BaseResource
-      # Constants...
-
-      # immutable # uncomment to make the resource immutable
-
-      # model_name / model_hint if required
-
       default_includes :uuid_object, :custom_metadata
 
-      # Associations:
-
+      ###
       # Attributes
-      attribute :uuid, readonly: true
-      attribute :metadata
+      ###
 
-      # This is required for POST
+      # @!attribute [r] uuid
+      #   @return [String] The UUID of the collection.
+      attribute :uuid
```

Comment:
Should we add `readonly: true` even though its excluded from the creatable and updatable fields. Means you dont need the creatable_fields method.

Reply:
Either way is legitimate.  The difference is in the error you get from the server.  I preferred the one where you use creatable and updatable fields but I'm happy to take advice.  I think it's better to avoid causing 500 server errors and the error message is much more technical when you get that response.

## Using `creatable_fields`
![image](https://github.com/user-attachments/assets/13bfcb6a-b645-4218-84c1-ab6cfa6769a0)

## Using `readonly: true`
![image](https://github.com/user-attachments/assets/e1830fd1-fa45-40fc-8da0-6904bb6b5e2a)

Reply:
Ah ok, yeah the first error is nicer. Happy for you to leave as is then.